st2api/st2api/controllers/v1/actionalias.py

```diff
@@ -104,80 +97,85 @@ def match(self, action_alias_match_api, **kwargs):
                                                     matches=[],
                                                     command=command)
             return [self._match_tuple_to_dict(match) for match in matches]
-        except (ActionAliasAmbiguityException) as e:
+        except ActionAliasAmbiguityException as e:
             LOG.exception('Command "%s" matched (%s) patterns.', e.command, len(e.matches))
-            pecan.abort(http_client.BAD_REQUEST, str(e))
-            return [self._match_tuple_to_dict(match) for match in e.matches]
+            return abort(http_client.BAD_REQUEST, str(e))
 
-    @request_user_has_permission(permission_type=PermissionType.ACTION_ALIAS_HELP)
-    @jsexpose(arg_types=[str, str, int], body_cls=ActionAliasHelpAPI,
-    status_code=http_client.ACCEPTED)
-    def help(self, action_alias_help_api, **kwargs):
+    def help(self, filter, pack, limit, offset, **kwargs):
```

Comment:
Nope, they absolutely should never be global. Pecan tried to play with global variables and we end up with things like get_requester() when you absolutely can not change the logic of the method without passing variable 10 levels down the stack.

Global variables are awful way of writing code and should be avoided unless absolutely required (read singleton).